nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -174,6 +177,18 @@ ProtectedNiFiProperties readProtectedPropertiesFromDisk(File file) {
             rawProperties.load(inStream);
             logger.info("Loaded {} properties from {}", rawProperties.size(), file.getAbsolutePath());
 
+            // Trim whitespace from each property. If property is multi-line, remove anything after the first line break.
```

Comment:
Should we allow for whitespace at the start/end of properties if the user wants it (e.g. a password)? So either a flag (in the properties) to enable/disable the whitespace trimming or an escape/quote megabits,  e.g. `"foo "` would retain the trailing whitespace but not the quotes?

Suspect this is somewhat of an edge case but always running whitespace can lead to frustrations

Reply:
Allow me to counterpoint:
1. Leading whitespace is taken care of by the underlying [java.util.Properties ](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#load-java.io.Reader-). That's already being stripped out and this pull request doesn't change that. So to support leading whitespace would mean writing a whole new reader.
2. Similarly, trailing whitespace on the _key_ is already taken care of by the Properties instance. So supporting leading whitespace would mean the possibility of "foo=bar" and "foo = bar" (which are currently equivalent) being different. And "foo =bar" would be equivalent to the former and not the latter. If nothing else, "supportLeadingWhitespace=false" should be the default, because you just know someone out there has configured their properties with spaces around the "=" and it's working now.
3. Supporting surrounding quote characters just changes the situation from "no trailing whitespace" to "no leading or trailing quote marks", though I supposed you could have a quote inside the value if you escape it somehow.

Reply:
Yeah, I realised the quotes issue after mashing the comment (i.e. just chasing it to handle surrounding quotes instead of whitespace)

Allowing for multi-line properties is good and as I say, this would be something of an edge case that isn't likely to happen regularly. If the standard property reader already removes such things anyway then as you say, maybe better to leave as-is